internal/app/feed/feed_inspection.go

```diff
@@ -181,25 +200,55 @@ func (f *InspectionFeed) Export(ctx context.Context, apiClient *api.Client, expo
 			Limit:         f.Limit,
 			WebReportLink: f.WebReportLink,
 		},
-	}, func(resp *api.GetFeedResponse) error {
-		rows := []*Inspection{}
+	}
+	feedFn := func(resp *api.GetFeedResponse) error {
+		var rows []Inspection
 
 		err := json.Unmarshal(resp.Data, &rows)
 		util.Check(err, "Failed to unmarshal inspections data to struct")
 
 		if len(rows) != 0 {
 			err = f.writeRows(exporter, rows)
-			util.Check(err, "Failed to write data to exporter")
+			if err != nil {
+				return err
+			}
 		}
 
-		logger.Info(GetLogString(f.Name(), &LogStringConfig{
+		lg.Info(GetLogString(f.Name(), &LogStringConfig{
 			RemainingRecords: resp.Metadata.RemainingRecords,
 			HTTPDuration:     apiClient.Duration,
 			ExporterDuration: exporter.GetDuration(),
 		}))
 		return nil
-	})
-	util.Check(err, "Failed to export feed")
+	}
 
-	return exporter.FinaliseExport(f, &[]*Inspection{})
+	return apiClient.DrainFeed(ctx, &req, feedFn)
+}
+
+func (f *InspectionFeed) processDeletedInspections(ctx context.Context, apiClient *api.Client, exporter Exporter) error {
+	lg := util.GetLogger()
+	dreq := api.NewGetAccountsActivityLogRequest(f.Limit, f.ModifiedAfter)
+	delFn := func(resp *api.GetAccountsActivityLogResponse) error {
+		var pkeys = make([]string, 0, len(resp.Activities))
+		for _, a := range resp.Activities {
+			uid := getPrefixID(a.Metadata["inspection_id"])
```

Comment:
Nit: can we name this `toS12ID` to match our internal function naming. This makes it a bit easier to SC Developers to recognise this function.

Reply:
done